File: PY-SDK/admin/archive/helpers/aver_client_setup_tests.py
from jsonrpcclient import requests
from solana.keypair import Keypair
from pyaver.aver_client import AverClient
from solana.rpc.commitment import Confirmed
from .token_airdrop import request_token_airdrop

async def aver_client_setup_tests(owner: Keypair, client: AverClient):
    #TEST AVER CLIENT
    #assert owner.public_key.to_base58() == client.owner.public_key.to_base58(), 'Test aver client'
    assert client.program is not None, 'Test aver client'

    #HEALTH CHECK
    health_check_result = await client.check_health()
    assert health_check_result.get('api') is not None, 'health check'
    assert health_check_result.get('solana') is not None, 'health check'

    #AIRDROP SOL
    sig = await client.provider.connection.request_airdrop(owner.public_key, 1_000_000_000, Confirmed)
    con = await client.provider.connection.confirm_transaction(sig['result'], Confirmed)
    balance = await client.provider.connection.get_balance(owner.public_key, Confirmed)
    assert(balance['result']['value'] == 1_000_000_000), 'airdrop sol'
    
    #AIRDROP USDC
    ata = await client.get_or_create_associated_token_account(
        owner.public_key,
        owner,
    )
    assert(ata is not None), 'get or create ATA'
    sig_2 = request_token_airdrop(client.aver_api_endpoint, client.quote_token,owner.public_key, 1_000_000_000)['signature']
    await client.provider.connection.confirm_transaction(sig_2, Confirmed) 
    token_balance = await client.request_token_balance(client.quote_token, owner.public_key)
    assert(token_balance == 1_000_000_000), 'token balance check after airdrop'

File: TS-SDK/admin/archive/__tests__/integration/utils/aver-client-setup-tests.ts
import { Keypair, Connection, LAMPORTS_PER_SOL } from '@solana/web3.js'
import { AverClient, signAndSendTransactionInstructions } from 'aver-ts'
import { AxiosResponse } from 'axios'
import { createAverClient } from './client'

export const getAverClientSetupTests = (owner: Keypair) => {
  describe('aver client setup tests', () => {
    let client: AverClient
    let keypair: Keypair
    let conn: Connection

    beforeAll(async () => {
      client = (await createAverClient(owner)) as AverClient
      keypair = owner
      conn = client.connection
    })

    test('test aver client', async () => {
      expect(client.owner.toBase58()).toBe(keypair.publicKey.toBase58())
      expect(client.program).toBeTruthy()
    })

    describe('health checks', () => {
      test('Health check is successful', async () => {
        const healthCheckResult = await client.checkHealth()
        expect(healthCheckResult.api).toBe(true)
        expect(healthCheckResult.solana).toBe(true)
      })
    })

    test('successfully airdrop Sol', async () => {
      const sig = await conn.requestAirdrop(keypair.publicKey, LAMPORTS_PER_SOL * 2)
      await client.connection.confirmTransaction(sig)
      const balance = await conn.getBalance(keypair.publicKey)
      expect(balance).toBe(LAMPORTS_PER_SOL * 2)
    })

    test('successfully airdrop USDC', async () => {
      // create ATA and wait for confirmation
      const ix = await client.createTokenAtaInstruction()
      const sig = await signAndSendTransactionInstructions(client.connection, [], keypair, [ix])
      const confirmedTx1 = await client.connection.confirmTransaction(sig)
      expect(confirmedTx1.value.err).toBeFalsy()

      const { signature } = ((await client.requestTokenAirdrop()) as AxiosResponse).data
      const confirmedTx2 = await client.connection.confirmTransaction(signature)
      expect(confirmedTx2.value.err).toBeFalsy()
      const newBalance = parseInt((await client.requestTokenBalance()).amount)

      expect(newBalance).toBe(1000 * Math.pow(10, 6))
    })
  })
}
